library/std/src/net/ip.rs

```diff
@@ -1131,10 +1339,22 @@ impl Ipv6Addr {
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0xffff, 0xc00a, 0x2ff).is_unspecified(), false);
     /// assert_eq!(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 0).is_unspecified(), true);
     /// ```
+    ///
+    /// # Stability guarantees
+    ///
+    /// This method's behavior [may by subject to changes][changes] in the
+    /// future, as new RFCs are published.
+    ///
+    /// [changes]: ../net/index.html#stability-guarantees-for-ietf-defined-behavior
     #[rustc_const_unstable(feature = "const_ipv6", issue = "76205")]
     #[stable(since = "1.7.0", feature = "ip_17")]
     pub const fn is_unspecified(&self) -> bool {
-        u128::from_be_bytes(self.octets()) == u128::from_be_bytes(Ipv6Addr::UNSPECIFIED.octets())
+        if let Some(v4_addr) = self.to_ipv4() {
```

Comment:
I don't _think_ this is a behavioral change, because an unspecified address is also just `0.0.0.0`.